api/envoy/data/dns/v3/dns_table.proto

```diff
@@ -28,22 +28,19 @@ message DnsTable {
     option (udpa.annotations.versioning).previous_message_type =
         "envoy.data.dns.v2alpha.DnsTable.AddressList";
 
-    // This field contains a well formed IP address that is returned
-    // in the answer for a name query. The address field can be an
-    // IPv4 or IPv6 address. Address family detection is done automatically
-    // when Envoy parses the string. Since this field is repeated,
-    // Envoy will return one randomly chosen entry from this list in the
-    // DNS response. The random index will vary per query so that we prevent
-    // clients pinning on a single address for a configured domain
+    // This field contains a well formed IP address that is returned in the answer for a
+    // name query. The address field can be an IPv4 or IPv6 address. Address family
+    // detection is done automatically when Envoy parses the string. Since this field is
+    // repeated, Envoy will return as many entries from this list in the DNS response while
+    // keeping the response under 512 bytes
```

Comment:
Sorry, probably an unrelated question, 512 bytes limit here is related to preventing DNS amplification attack?

Reply:
No, the 512 byte is imposed by the RFC.   There may be transport layer considerations that contribute to this limit.

My understanding of a DNS amplification attack is that the query is spoofed to appear to originate from the target DNS server's address.  Capping the message size wouldn't really be a mitigation here.   Rate limiting appears to be a good countermeasure to an amplification attack.  We would have to configure other filters to handle this type of traffic pattern.  Worst case iptables could probably be used to limit the incoming udp sessions.

Doing this rate limiting in the filter would require it to keep state.   I'm not certain we want to tackle that in this initial implementation of the filter.